Formula/zrepl.rb

```diff
@@ -0,0 +1,84 @@
+class Zrepl < Formula
+  desc "One-stop ZFS backup & replication solution"
+  homepage "https://zrepl.github.io"
+  url "https://github.com/zrepl/zrepl/archive/v0.1.1.tar.gz"
+  sha256 "0c16554e4527d14a390d78cf95bce759da425019a83ec63acfed5b4c50d68c9c"
+  head "https://github.com/zrepl/zrepl.git"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+  def install
+    contents = Dir["{*,.git,.gitignore}"]
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/zrepl/zrepl").install contents
+
+    ENV["GOPATH"] = gopath
+    ENV.prepend_create_path "PATH", gopath/"bin"    
+    cd gopath/"src/github.com/zrepl/zrepl" do
+      system "dep", "ensure", "-v", "-vendor-only"
+      system "go", "build", "-o", "'$GOPATH/bin/stringer'", "./vendor/golang.org/x/tools/cmd/stringer"
+      system "go", "build", "-o", "'$GOPATH/bin/protoc-gen-go'", "./vendor/github.com/golang/protobuf/protoc-gen-go"
+      system "go", "build", "-o", "'$GOPATH/bin/enumer'", "./vendor/github.com/alvaroloes/enumer"
+      system "go", "build", "-o", "'$GOPATH/bin/goimports'", "./vendor/golang.org/x/tools/cmd/goimports"
+      system "go", "build", "-o", "'$GOPATH/bin/golangci-lint'", "./vendor/github.com/golangci/golangci-lint/cmd/golangci-lint"
+      system "make", "ZREPL_VERSION=#{version}"
+      bin.install "artifacts/zrepl"
+    end      
+  end
+
+  def post_install
+    (var/"log/zrepl").mkpath
+    (var/"run/zrepl").mkpath
+    (etc/"zrepl").mkpath
+  end
+
+  plist_options :startup => true, :manual => "sudo zrepl daemon"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN"
+    "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>EnvironmentVariables</key>
+        <dict>
+          <key>PATH</key>
+          <string>/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:#{HOMEBREW_PREFIX}/bin</string>
```

Comment:
These last 2 should be the same.

Reply:
so you want me to remove /usr/local/bin from the path? Or homebrew_prefix? Some people might use a different prefix

Reply:
/usr/local/bin

Reply:
I'm not sure but zrepl might need to have the zfs and zpool binaries in its $PATH when run. Those binaries are installed into /usr/local/bin. If the homebrew_prefix is a different one, zrepl would be unable to access them. Or what am I missing here? (Really not experienced with the small details, please let me know)

Reply:
I guess it makes more sense then. But that would mean that it depends on those on runtime correct?

Reply:
I think so. But zfs and zpool are not and cannot reasonably be integrated automatically into homebrew. In that sense, it's pretty much exactly the same as the `znapzend` formula already merged and existing in homebrew.